### PR TITLE
Handling of descriptions

### LINKED ISSUE #52: Groups should have a description

### PR DESCRIPTION
**Pull request by britter**
Descriptions are now rendered into the the JavaDoc of getters.

Resolves #52, #53

### CODE REVIEW

File: src/main/java/org/gradlex/buildparameters/PluginCodeGeneration.java [hunk 1]
```diff
@@ -29,7 +29,9 @@
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Iterator;
```
Comment (britter) [inline src/main/java/org/gradlex/buildparameters/PluginCodeGeneration.java:32]: unused

Comment (jjohannes) [reply src/main/java/org/gradlex/buildparameters/PluginCodeGeneration.java:32]: ? Why not remove it then?

**Review by jjohannes — APPROVED** [review]
Very good 👍 

We should update the documentation about groups as well (add the description). This can be a follow up.

I tried this on a project and one thing I noticed, which is only indirectly related, is that if you navigate into the code to see the description, you also see all this implementation details clutter (like the injecting constructor). E.g.:

```

public abstract class Catalina {
    private final org.gradle.api.provider.Provider<String> home;
    @Inject
    public Catalina(ProviderFactory providers, ObjectFactory objects) {
        this.home = providers.gradleProperty("catalina.home").orElse(providers.environmentVariable("CATALINA_HOME"));
    }
    /**
     * Root of the Tomcat installation
     */
    public org.gradle.api.provider.Provider<String> getHome() {
        return this.home;
    }
}
```

We could consider generating interfaces for the extension and the group and work only with these in the type signatures. And then the JavaDoc would also go only there.

File: src/main/java/org/gradlex/buildparameters/PluginCodeGeneration.java (hunk 1 above)
Comment (britter) [reply src/main/java/org/gradlex/buildparameters/PluginCodeGeneration.java:32]: Ran out of time yesterday and added this comment as a reminder 😇

**Pull request merged**